fail autopkg hard if GitHubReleasesInfoProvider.py queries an archived repo, unless fail_if_archived is False

discentem opened the pull request.

discentem commented on Code/tests/test_githubreleasesinfoprovider.py:
```diff
@@ -74,7 +76,28 @@ def test_returns_asset_url(self):
         self.processor.main()
         self.assertIsNotNone(test_env["asset_url"])
+    
+    @patch('autopkglib.GitHubReleasesInfoProvider.is_archived')
```
Please let me know if I should add more tests elsewhere

discentem commented:
Not sure why `TestGithubReleaseInfoProvider (macos-latest)` is failing 😞

homebysix commented:
I like this concept, but I have a nitpick about this mismatch between the input variable name (`ignore_archived`) and its behavior (raises an exception, rather than "ignoring"). I don't think it's desirable for an exception to be raised, since even archived repositories can successfully provide software download assets.

I could envision something similar to DeprecationWarning, where GitHubReleasesInfoProvider raises a warning in the recipe summary output to notify the AutoPkg user that the GitHub repo is archived, like so:

```
The following recipes raised GitHub release warnings:
    Name          Warning
    ----          -------
    Foo.download  The GitHub repository hosting assets for this recipe has been
                  archived, and is therefore unlikely to provide future updates.
```

Unfortunately there's no common "warnings" summary output for AutoPkg; it's up to each processor to determine what it adds to its `*_summary_result` variables. So this would be a new `github_releases_summary_result` output variable, probably.

discentem commented:
Thanks for the reply @homebysix! I implemented it this way due to Nick's comment here: https://github.com/autopkg/autopkg/issues/934#issuecomment-1934576200

However, I'm happy to change the variable name, reverse the behavior, and/or switch to a warning instead of ProcessorError if that is the preferred direction.

discentem commented:
Oh, I definitely made a mistake with my current implementation. But only in the description I think?

1. The description is wrong:

```diff
"description": (
- "If set to True, the processor will raise an exception if the "
-  "GitHub repo is archived."
+  "If set to False, the processor will raise an exception if the Github repo is archived."
),
```

nmcspadden commented:
Is this something you're still interested in fixing?

discentem commented:
> Is this something you're still interested in fixing?

Yeah, sure thing! Other than updating the description, are there any changes you want to see? Also if you have any advice about the failing test that would be very helpful.

discentem committed:
Implement PR 961: archived repo detection with ignore_archived option

- Add ignore_archived input variable to GitHubReleasesInfoProvider
- Add get_repo() method to fetch repo metadata from GitHub API
- Add is_archived() method to check if repo is archived
- Update main() to check archived status before fetching releases
- Optimize to skip API call if ignore_archived is already set
- Add comprehensive tests using patch.object()
- Follow Copilot review suggestions for code quality

discentem committed:
Use fail_if_archived variable with True as default

- fail_if_archived is True by default (fail on archived repos)
- Set to False to allow archived repos
- Updated error message with clearer instructions
- Updated test to match the new behavior

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci
